main.go

```diff
@@ -69,6 +70,10 @@ func main() {
 			os.Exit(1)
 		}
 
+		if newLine := strings.Index(string(secretKeyBytes), "\n"); newLine != -1 {
```

Comment:
Let's cast to string only once, store it in a variable and then re-use it. In fact it would be ideal to extract a small function for this purpose. It can then be unit tested too.

Thanks,

Alex

Reply:
Unit tests should show positive/negative case and what happens if no bytes are read - i.e. 0 length string